nvflare/app_opt/job_launcher/k8s_launcher.py

```diff
@@ -0,0 +1,263 @@
+                "-t": None,
+                "-d": None,
+                "-n": None,
+                "-c": None,
+                "-p": None,
+                "-g": None,
+                "-scheme": None,
+                "-s": None,
+            },
+        )
+        self.container_args_module_args_dict_as_list = list()
+        for k, v in self.container_args_module_args_dict.items():
+            self.container_args_module_args_dict_as_list.append(k)
+            self.container_args_module_args_dict_as_list.append(v)
+        self.volume_list = job_config.get(
+            "volume_list", [{"name": None, "hostPath": {"path": None, "type": "Directory"}}]
+        )
+
+        self.pod_manifest["metadata"]["name"] = job_config.get("name")
+        self.pod_manifest["spec"]["containers"] = self.container_list
+        self.pod_manifest["spec"]["volumes"] = self.volume_list
+
+        self.container_list[0]["image"] = job_config.get("image", "nvflare/nvflare:2.5.0")
+        self.container_list[0]["name"] = job_config.get("container_name", "nvflare_job")
+        self.container_list[0]["args"] = (
+            self.container_args_python_args_list
+            + self.container_args_module_args_dict_as_list
+            + self.container_args_module_args_sets
+        )
+        self.container_list[0]["volumeMounts"] = self.container_volume_mount_list
+
+    def get_manifest(self):
+        return self.pod_manifest
+
+    def enter_states(self, job_states_to_enter: list, timeout=None):
+        starting_time = time.time()
+        if not isinstance(job_states_to_enter, (list, tuple)):
+            job_states_to_enter = [job_states_to_enter]
+        if not all([isinstance(js, JobState)] for js in job_states_to_enter):
+            raise ValueError(f"expect job_states_to_enter with valid values, but get {job_states_to_enter}")
+        while True:
+            job_state = self.poll()
+            if job_state in job_states_to_enter:
+                return True
+            elif timeout is not None and time.time() - starting_time > timeout:
+                return False
+            time.sleep(1)
+
+    def terminate(self):
+        resp = self.api_instance.delete_namespaced_pod(
+            name=self.job_id, namespace=self.namespace, grace_period_seconds=0
+        )
+        return self.enter_states([JobState.TERMINATED], timeout=self.timeout)
+
+    def poll(self):
+        try:
+            resp = self.api_instance.read_namespaced_pod(name=self.job_id, namespace=self.namespace)
+        except ApiException as e:
+            return JobState.UNKNOWN
+        return POD_STATE_MAPPING.get(resp.status.phase, JobState.UNKNOWN)
+
+    def wait(self):
+        self.enter_states([JobState.SUCCEEDED, JobState.TERMINATED])
+
+
+class K8sJobLauncher(JobLauncherSpec):
+    def __init__(
+        self,
+        config_file_path,
+        root_hostpath: str,
+        workspace: str,
+        mount_path: str,
+        supported_images: [str] = None,
```

Comment:
This is supposed to be provided by local IT, but how can the IT person know it?

Reply:
removed supported_images.